src/devices/Ad7193/Ad7193.cs

```diff
@@ -0,0 +1,855 @@
+
+            _stopWatch.Restart();
+        }
+
+        /// <summary>
+        /// Initiates Continuous Conversion. After calling this method the AdcValueReceived event will be fired continuously. This mode is much faster, but uses significatily more CPU power.
+        /// </summary>
+        /// <param name="frequency">The target frequency of the sampling. AD7193 has the maximum of 4800 samples per second.</param>
+        public void StartContinuousConversion(uint frequency = 0)
+        {
+            if (frequency == 0)
+            {
+                frequency = metadata.ADCSamplerate;
+            }
+
+            if (frequency > metadata.ADCSamplerate)
+            {
+                throw new ArgumentException($"The frequency you provided is higher than the maximum sampling rate of AD7193 ({metadata.ADCSamplerate} SPS).");
+            }
+
+            _registerCache[(byte)Register.Mode] = SetRegisterBits(Register.Mode, BitMask.ModeMD, 0b000);      // continuous conversion mode
+            SetRegisterValue(Register.Mode, _registerCache[(byte)Register.Mode]);
+
+            ContinuousRead = true;
+
+            long samplePerTicks = Stopwatch.Frequency / frequency;
+            if (samplePerTicks == 0)
+            {
+                samplePerTicks = 1;
+            }
+
+            _stopWatch.Restart();
+            Task samplingTask = Task.Run(() =>
+            {
+                long samples = 0;
+                long nextSampleAt = _stopWatch.ElapsedTicks;
+                long elapsedTicks = 0;
+                long jitter = 0;
+                long maxJitterCorrectionPerSample = Math.Max(Stopwatch.Frequency / 100000, 1);
+                while (_stopWatch.IsRunning)
+                {
+                    elapsedTicks = _stopWatch.ElapsedTicks;
+                    if (elapsedTicks >= nextSampleAt)
+                    {
+                        nextSampleAt = elapsedTicks + samplePerTicks;
+                        ReadADCValue();
+                        samples++;
+                        jitter = (elapsedTicks - (samples * samplePerTicks));
+                        if (JitterCorrection)
+                        {
+                            if (jitter > 0)
+                            {
+                                nextSampleAt -= Math.Min(jitter, maxJitterCorrectionPerSample);
+                            }
+                            else if (jitter < 0)
+                            {
+                                nextSampleAt += Math.Min(-jitter, maxJitterCorrectionPerSample);
+                            }
+                        }
+                    }
+                }
+            });
+        }
+
+        /// <summary>
+        /// Reads the current ADC result value on the selected channel
+        /// </summary>
+        /// <returns>24-bit raw value of the last ADC result (+ status byte if enabled)</returns>
+        public uint? ReadADCValue()
```

Comment:
We usually use TryReadValue/TryReadADCValue instead of nullable. Also if there can be multiple types of error and not just "error" you might want to return status or just throw an exception if this is super unlikely

Reply:
I checked and it can be only one general error, if the Status register indicates it.
I need to test if it is common to have a few errors during conversions, I'll do it today.

Reply:
Alright, so in order to get an error, you must misconfigure something, otherwise the conversion goes on without any errors even at the maximum sampling rate.
Because of this, I changed the return value to a non-nullable uint and added a logic to throw an Exception if something goes wrong. It should happen fairly rarely.

Reply:
I did some further testing, and I was wrong: the error bit gets set even if the signal on the ADC channel is out of range. Because of this, I changed the method to TryReadADCValue(out uint value) returning a bool.